### ISSUE TITLE
Create a "CWL linter" tool

### ISSUE THREAD

**Issue by melibleq** [common-workflow-language/common-workflow-language#935]
Following a suggestion from @kinow, the idea is to have a `cwl-linter module` to inspect workflows and detect possible improvements in the code.

Related: https://github.com/common-workflow-lab/wdl-cwl-translator/issues/127 and https://github.com/rabix/cwl-format/issues/18

**Comment by kinow** [comment]
And maybe have something that can re-write worfklows too. Similar to Python's `black`, or `isort` for imports. It could sort the sections of a CWL document, e.g.: https://github.com/common-workflow-lab/wdl-cwl-translator/pull/172/files#diff-b5a65c2befbbd0b6453805272729b622f41b2be683f8258432e08f4370332265R83

**Comment by mr-c** [comment]
@kinow , yep, content reordering is already a feature of `cwl-format` https://github.com/rabix/cwl-format/blob/master/cwlformat/keyorder.yml

`cwl-format` hasn't had a release in a while, though.
